src/components/Loader.vue

```diff
@@ -167,27 +181,37 @@ class Loader extends Vue {
         }
 
         if (newState === Loader.State.LOADING) {
-            if (this.stateUpdateTimeout) {
+            if (this.stateUpdateTimeout !== -1) {
                 clearTimeout(this.stateUpdateTimeout);
-                delete this.stateUpdateTimeout;
+                this.stateUpdateTimeout = -1;
             }
-            if (this.indicatorDisplayTimeout) {
+            if (this.indicatorDisplayTimeout !== -1) {
                 clearTimeout(this.indicatorDisplayTimeout);
-                delete this.indicatorDisplayTimeout;
+                this.indicatorDisplayTimeout = -1;
             }
             this.showLoadingBackground = true;
             this.strokeDashoffset = Loader.STROKE_DASHOFFSET;
         }
     }
 
     @Watch('status', {immediate: true})
-    private updateStatus(newStatus: string, oldStatus: string) {
+    private async updateStatus(newStatus: string) {
+        const loadingStatusRow = this.$refs.loadingStatusRow as HTMLElement;
+        if (this.statusUpdateTimeout !== -1) {
+            // reset transitioning state for new change
+            loadingStatusRow.classList.remove('transition');
+            loadingStatusRow.offsetWidth; // tslint:disable-line:no-unused-expression
```

Comment:
Hum, I don't think it's a hack. It's one of the better options to enforce a style update.
A style update needs to be enforced here to reset / restart the transition. Otherwise, the browser would note note that the class was removed and readded.

Instead of `.offsetWidth` using `await new Promise(resolve => requestAnimationFrame(resolve));` might also work.